test/e2e/operator_test.go

```diff
@@ -70,172 +76,175 @@ var _ = Describe("Operator API", func() {
 
-		// Delete o
-		Eventually(func() error {
-			err := client.Delete(clientCtx, o)
-			if err != nil && !apierrors.IsNotFound(err) {
-				return err
-			}
-			return nil
-		}).Should(Succeed())
+			// Delete o
+			Eventually(func() error {
+				err := client.Delete(clientCtx, o)
+				if err != nil && !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}).Should(Succeed())
 
-		// Ensure that o is eventually recreated (because some of its components still exist).
-		By("recreating the Operator when any components still exist")
-		Eventually(func() error {
-			return client.Get(clientCtx, types.NamespacedName{Name: o.GetName()}, o)
-		}).Should(Succeed())
+			// Ensure that o is eventually recreated (because some of its components still exist).
+			By("recreating the Operator when any components still exist")
+			Eventually(func() error {
+				return client.Get(clientCtx, types.NamespacedName{Name: o.GetName()}, o)
+			}).Should(Succeed())
 
-		// Delete ns-a
-		Eventually(func() error {
-			err := client.Delete(clientCtx, nsA)
-			if apierrors.IsNotFound(err) {
-				return nil
-			}
-			return err
-		}).Should(Succeed())
+			// Delete ns-a
+			Eventually(func() error {
+				err := client.Delete(clientCtx, nsA)
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}).Should(Succeed())
 
-		// Ensure the reference to ns-a is eventually removed from o's status.components.refs field
-		By("removing a component's reference when it no longer exists")
-		componentRefEventuallyExists(w, false, getReference(scheme, nsA))
+			// Ensure the reference to ns-a is eventually removed from o's status.components.refs field
+			By("removing a component's reference when it no longer exists")
+			componentRefEventuallyExists(w, false, getReference(scheme, nsA))
 
-		// Delete o
-		Eventually(func() error {
-			err := client.Delete(clientCtx, o)
-			if apierrors.IsNotFound(err) {
-				return nil
-			}
-			return err
-		}).Should(Succeed())
+			// Delete o
+			Eventually(func() error {
+				err := client.Delete(clientCtx, o)
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}).Should(Succeed())
 
-		// Ensure that o is consistently not found
-		By("verifying the Operator is permanently deleted if it has no components")
-		Consistently(func() error {
-			err := client.Get(clientCtx, types.NamespacedName{Name: o.GetName()}, o)
-			if apierrors.IsNotFound(err) {
-				return nil
-			}
-			return err
-		}).Should(Succeed())
+			// Ensure that o is consistently not found
+			By("verifying the Operator is permanently deleted if it has no components")
+			Consistently(func() error {
+				err := client.Get(clientCtx, types.NamespacedName{Name: o.GetName()}, o)
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}).Should(Succeed())
+		})
 	})
 
 	Context("when a subscription to a package exists", func() {
```

Comment:
Note: we were previously leaking the kiali installation, it's CRDs, and the corresponding Operator resource that was generated despite these test cases passing or failing.